help-docs: Note automated PMs sent when subscribing users to streams.

laurynmm opened the pull request:
Adds a note to both help center articles with sections on subscribing users to streams, documenting that an automated private message (from the Zulip notification bot) is sent to each user when subscribed in this way.

Also, makes a small formatting change to the existing note [here](https://zulip.com/help/manage-user-stream-subscriptions#view-a-users-stream-subscriptions), see 3rd screenshot below.

Noted while working on #22612.

**Screenshots and screen captures:**

[Current documentation](https://zulip.com/help/manage-user-stream-subscriptions#subscribe-a-user-to-a-stream)
![Screenshot from 2022-07-28 16-44-37](https://user-images.githubusercontent.com/63245456/181749367-2cd0a4e7-f659-4365-95e8-f228d46d3d04.png)

[Current documentation](https://zulip.com/help/add-or-remove-users-from-a-stream#add-users-to-a-stream)
![Screenshot from 2022-07-28 16-44-53](https://user-images.githubusercontent.com/63245456/181749372-62b7a311-320c-4b4c-bc4c-e529b8ba5b71.png)

[Current documentation](https://zulip.com/help/manage-user-stream-subscriptions#view-a-users-stream-subscriptions)
![Screenshot from 2022-07-29 13-30-44](https://user-images.githubusercontent.com/63245456/181750086-87c48a4e-822f-4617-86fd-a2535529e473.png)


**Self-review checklist**

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/version-control.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.

alya commented:
Nice! For the wording, maybe something like this?

> Note: Subscribing someone else to a stream sends them an automated private message.

I think that hints the fact that you don't get a message when you subscribe yourself through this part of the UI, and would also let us use the same note in both places.

alya commented:
Everything else looks good to me!

laurynmm commented:
**Update**: I updated the wording with a slight adjustment to Alya's suggestion above (see quote and screenshots below):

> Note: Subscribing someone else to a stream sends them a **notification bot** automated private message.

I felt like highlighting that the automated private message would be from the notification bot, and not the user making the change, was a good idea. Also, I linked to the notification bot article in the note and updated that article to be clearer in text related to automated private messages (see 3rd screenshot below).

**Updated screenshots**:

[Current documentation](https://zulip.com/help/add-or-remove-users-from-a-stream)
![Screenshot from 2022-08-08 11-07-22](https://user-images.githubusercontent.com/63245456/183389691-3fa5391f-5f82-4a3a-ba79-534c9c4153d9.png)

[Current documentation](https://zulip.com/help/manage-user-stream-subscriptions)
![Screenshot from 2022-08-08 11-06-57](https://user-images.githubusercontent.com/63245456/183389781-a7dc2c68-761a-4d7c-8370-9aaa1bd31e9a.png)

[Current documentation](https://zulip.com/help/configure-notification-bot)
![Screenshot from 2022-08-08 11-48-06](https://user-images.githubusercontent.com/63245456/183390285-001ea122-0170-40cd-b002-77db72caec5f.png)

timabbott commented:
What do you think about "... sends them an automated private message from Notification Bot."? I think that's most clear.

laurynmm commented:
> What do you think about "... sends them an automated private message from Notification Bot."? I think that's most clear.

That makes sense. I've updated this PR for the above text with one small change - not capitalizing notification bot (see screenshot below), as that's the direction we've been moving with the help center docs for the notification bot in general (see [current documentation](https://zulip.com/help/configure-notification-bot)).

It could be that we want to revisit that and keep all mentions of the notification bot in the help center to be capitalized (e.g. Notification Bot) with the argument that it's how the bot appears in the UI.

**Screenshot of updated warning note**:
![Screenshot from 2022-08-15 11-38-15](https://user-images.githubusercontent.com/63245456/184618220-4bc19774-d703-4333-bc7b-6cd8dedade68.png)

timabbott commented:
Merged, thanks @laurynmm! Yeah probably worth discussing how we want to capitalize it; it needs to be Notification Bot in the UI because lower-case names are not the norm in a Zulip community, and that capitalization choice helps signal that. But I can imagine that story changing if we rename it; in any case, sticking with whatever our current convention is seems right for this PR :).
